packages/ember-data/lib/system/relationships/state/has-many.js

```diff
@@ -144,6 +145,9 @@ ManyRelationship.prototype.computeChanges = function(records) {
 ManyRelationship.prototype.fetchLink = function() {
   var self = this;
   return this.store.findHasMany(this.record, this.link, this.relationshipMeta).then(function(records) {
+    var meta = self.store.metadataFor(self.relationshipMeta.type);
+    self.manyArray.set('meta', cloneNull(meta));
```

Comment:
We have a potential problem here, if you make a request for a record, set some metadata, and then fetch the link, *do not set the metadata*, this array will end up with wrong metadata. Not sure how to fix this right now